### PR TITLE
updated doc with code sample for google analytics integration

### PR DESCRIPTION
**Pull request by sloppycoder**
the code was originally from exponent/ex-navigation repo. adapted slightly.

### CODE REVIEW

**Comment by satya164** [comment]
Thanks for the PR. Let's change the name of the file to `Screen tracking and analytics`

File: docs/guides/GoogleAnalytics-Integration.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+## Integration with GoogleAnalytics
+
+This example shows how to use a middleware to track navigation between screens and send to Google Analytics. The approach is generic, should work with any mobile analytics SDK.
+The code is originally from [exponent/ex-navigtation](https://github.com/exponent/ex-navigation) repo, only adapted slightly.
```
Comment (satya164) [inline docs/guides/GoogleAnalytics-Integration.md]: I think we don't need this line

File: docs/guides/GoogleAnalytics-Integration.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+### Create middleware
+
+```
```
Comment (satya164) [inline docs/guides/GoogleAnalytics-Integration.md]: add `js`

File: docs/guides/GoogleAnalytics-Integration.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  }
+  return result;
+};
+
+export default screenTracking;
+
+```
+
+### Create Redux store and apply the above middleware
+```
```
Comment (satya164) [inline docs/guides/GoogleAnalytics-Integration.md]: add `js`

**Comment by satya164** [comment]
We need to add a little more info about what happens in second two sections. 1-2 lines will be sufficient.

**Comment by sloppycoder** [comment]
updated. please review. what you mean by "add js"? sorry, newbie here.

**Comment by ericvicenti** [comment]
Can you add the doc to the website's `App.js` Please run the website to verify it works.

I agree with @satya164 that the doc should be renamed.

Also I think this should be revised to show how to add analytics without redux. The redux-specific parts of the guide should link to the redux guide as a prerequisite

**Comment by sloppycoder** [comment]
Updated doc and App.js. 
I don't know how to make it work without Redux, could you give me some hint? Can I still use the same middleware? How to apply that?

**Comment by ericvicenti** [comment]
If you see the redux guide, you'll realize that every router can simply take a navigation prop with a nav state, and a dispatch function. If you log on the `componentDidUpdate` of your top-level component, you can see the changing `navigation.state` and do logging from there

I'm hesitant to add a new guide that is specific to those using redux. So hopefully we can make this more generic.

**Comment by sloppycoder** [comment]
I guess I don't know React well enough to understand this. I added a few hooks to top level component, but for some reason that I don't understand, the componentDidUpdate and componentShouldUpdate are never called. Could you drop more hints pleases?


(Test Code Repo)[https://github.com/sloppycoder/ScreenTrackingTest.git]

```
const AppNavigator = StackNavigator(
  {
    dashboard: { screen: DashboardScreen },
    transaction: { screen: TransactionDetailScreen },
    login: { screen: LoginScreen },
  },
  {
    initialRouteName: 'login'
  }
);

class App extends Component {

   // this never gets called
    shouldComponentUpdate(nextProps, nextState) {
      console.log('App.shouldComponentUpdate', nextProps, nextState);
    }

  // this never gets called
    componentDidUpdate(prevProps, prevState) {
        console.log('App.componentDidUpdate', prevProps, prevState);
    }

   // only this one gets called. 
    componentDidMount() {
        console.log('App.componentDidMount');
    }

    render() {
        return (
            <AppNavigator/>
        );
    }
}

```

**Comment by sloppycoder** [comment]
ok. I figured it out. will update soon.

**Comment by sloppycoder** [comment]
Example without Redux added.

File: docs/guides/Screen-Tracking.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
+## Screen tracking and analytics
+
+This example shows how to do screen tracking and send to Google Analytics. The approach can be adapted to any other mobile analytics SDK. 
+
+### Use componentDidUpdate hook
+`componentDidUpdate` has access the previous and current navigation state and its a good place to do screen tracking.
+
+```
```
Comment (satya164) [inline docs/guides/Screen-Tracking.md]: Add `js`

File: docs/guides/Screen-Tracking.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
+## Screen tracking and analytics
+
+This example shows how to do screen tracking and send to Google Analytics. The approach can be adapted to any other mobile analytics SDK. 
+
+### Use componentDidUpdate hook
+`componentDidUpdate` has access the previous and current navigation state and its a good place to do screen tracking.
```
Comment (satya164) [inline docs/guides/Screen-Tracking.md]: New line after heading

File: docs/guides/Screen-Tracking.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+  const prevScreen = screenName(prevState.nav);
+  if (!!currScreen && currScreen != prevScreen) {
+    console.log('tracking screen', currScreen);
+    tracker.trackScreenView(currScreen);
+  }
+}
+```
+
+### Use Redux
+When using Redux, `screenTracking` can be written as a Redux middleware.
```
Comment (satya164) [inline docs/guides/Screen-Tracking.md]: New line after heading

File: docs/guides/Screen-Tracking.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+```
```
Comment (satya164) [inline docs/guides/Screen-Tracking.md]: Add js

File: docs/guides/Screen-Tracking.md [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+    console.log('tracking middleware: ', nextScreen);
+    // the line below uses the Google Analytics tracker
+    // change the tracker here to use other Mobile analytics SDK.
+    tracker.trackScreenView(nextScreen);
+  }
+  return result;
+};
+
+export default screenTracking;
+
```
Comment (satya164) [inline docs/guides/Screen-Tracking.md]: nit: remove empty line

File: docs/guides/Screen-Tracking.md [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+```
+
+### Create Redux store and apply the above middleware
+The `screenTracking` middleware can be applied to the store during its creation. See [Redux Integration](Redux-Integration.md) for details.
```
Comment (satya164) [inline docs/guides/Screen-Tracking.md]: nit: new line after heading

File: docs/guides/Screen-Tracking.md [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+```
```
Comment (satya164) [inline docs/guides/Screen-Tracking.md]: nit: new line before code block and add `js`

File: docs/guides/Screen-Tracking.md [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+  combineReducers({
+    nav: navReducer,
+    ...
+  }),
+  applyMiddleware(
+    screenTracking,
+    ...
+    ),
+);
+
```
Comment (satya164) [inline docs/guides/Screen-Tracking.md]: nit: remove new line

File: docs/guides/Screen-Tracking.md (hunk 7 above)
Comment (sloppycoder) [inline docs/guides/Screen-Tracking.md]: add js to where? this screen-tracking.md has been added to website/src/App.js

File: docs/guides/Screen-Tracking.md (hunk 10 above)
Comment (satya164) [inline docs/guides/Screen-Tracking.md]: When you write codeblock in markdown, you have to add `js` after the 3 backticks so that it's syntax highlighted

**Comment by sloppycoder** [comment]
add/removed empty lines and added js after 3 ticks to enable syntax highlighting. Thanks for the tips.

**Pull request merged**